src/runtime.js

```diff
@@ -228,7 +214,54 @@ function adoptShared(global, callback) {
  * @param {!Window} global Global scope to adopt.
  */
 export function adopt(global) {
-  adoptShared(global, global => {
+  adoptShared(global, (global, modules) => {
+    const ampdocService = ampdocFor(global);
+    const ampdoc = ampdocService.getAmpDoc(/* node */ null);
+
+    /**
+     * Registers an extended element and installs its styles.
+     * @param {string} name
+     * @param {!Function} implementationClass
+     * @param {string=} opt_css Optional CSS to install with the component.
+     *     Typically imported from generated CSS-in-JS file for each component.
+     */
+    global.AMP.registerElement = function(name, implementationClass, opt_css) {
+      registerModuleElement(modules, name, implementationClass);
+      const register = function() {
+        registerExtendedElement(global, name, implementationClass);
+        if (getMode().test) {
+          elementsForTesting.push({
+            name,
+            implementationClass,
+            css: opt_css,
+          });
+        }
+        // Resolve this extension's Service Promise.
+        getService(global, name, emptyService);
+      };
+      if (opt_css) {
+        installStyles(global.document, opt_css, register, false, name);
+      } else {
+        register();
+      }
+    };
+
+    /**
+     * Registers an ampdoc service.
+     * @param {string} name
+     * @param {function(new:!Object, !./service/ampdoc-impl/AmpDoc)} opt_class
+     * @param {function(!./service/ampdoc-impl/AmpDoc):!Object} opt_factory
+     */
+    global.AMP.registerDocService = function(name, opt_class, opt_factory) {
```

Comment:
Unfortunately, not yet ready to do services quite yet. They are still a bit of a web of dependencies. The only leaf extension I'm ready to migrate right now is amp access and, naturally, i'd like to do it a bit more carefully.